samcli/commands/validate/validate.py

```diff
@@ -95,3 +104,63 @@ def _read_sam_file(template):
         sam_template = yaml_parse(sam_template.read())
 
     return sam_template
+
+
+def _lint(ctx: Context, template: str) -> None:
+    """
+    Parses provided SAM template and maps errors from CloudFormation template back to SAM template.
+
+    Cfn-lint loggers are added to the SAM cli logging hierarchy which at the root logger
+    configures with INFO level logging and a different formatting. This exposes and duplicates
+    some cfn-lint logs that are not typically shown to customers. Explicitly setting the level to
+    WARNING and propagate to be False remediates these issues.
+
+    Parameters
+    -----------
+    ctx
+        Click context object
+    template
+        Path to the template file
+
+    """
+
+    import cfnlint.core  # type: ignore
+    import logging
+    from samcli.commands.exceptions import UserException
+    from botocore.utils import validate_region_name
+    from botocore.exceptions import InvalidRegionError
+
+    cfn_lint_logger = logging.getLogger("cfnlint")
+    cfn_lint_logger.propagate = False
+
+    try:
+        validate_region_name(ctx.region)
```

Comment:
Lets move this entire check to within the cli validator for `region`

```python
from samcli.commands.exceptions import RegionError

    def callback(ctx, param, value):
        state = ctx.ensure_object(Context)
        from botocore import exceptions, utils

        try:
            utils.validate_region_name(value)
        except exceptions.InvalidRegionError as ex:
            raise RegionError(
                message=f"Provided region: {value} doesn't match a supported format", wrapped_from=ex.__class__.__name__
            )
        state.region = value
        return value

    return click.option(
        "--region", expose_value=False, help="Set the AWS Region of the service (e.g. us-east-1).", callback=callback
    )(f)
```

at https://github.com/aws/aws-sam-cli/blob/develop/samcli/cli/options.py#L41

Reply:
@cdavidxu-hub I think it was a misunderstanding that I didn't want to keep as it is, instead I was suggesting to move this validation to a more generic place or removing it completely. Let's continue with what @sriram-mv suggested above.